src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -156,8 +158,77 @@ public ReplicationLoader(DocumentDatabase database, ServerStore server)
                 null, reconnectTime, reconnectTime);
             MinimalHeartbeatInterval = (int)config.ReplicationMinimalHeartbeat.AsTimeSpan.TotalMilliseconds;
             database.TombstoneCleaner.Subscribe(this);
+            server.Cluster.Changes.DatabaseChanged+=DatabaseValueChanged;
         }
 
+        private Task DatabaseValueChanged(string databaseName, long index, string type, DatabasesLandlord.ClusterDatabaseChangeType changeType, object changeState)
+        {
+            var documentDatabase = Database;
+            if (documentDatabase == null)
+                return Task.CompletedTask;
+            
+            if (string.Equals(documentDatabase.Name, databaseName, StringComparison.OrdinalIgnoreCase) == false)
+                return Task.CompletedTask;
+
+            switch (changeState)
+            {
+                case UpdatePullReplicationAsHubCommand put:
+                    DisposeRelatedPullReplication(put.Definition.Name, null /*all*/);
+                    break;
+                case UnregisterReplicationHubAccessCommand del:
+                    DisposeRelatedPullReplication(del.HubDefinitionName, del.CertThumbprint);
+                    break;
+                case RegisterReplicationHubAccessCommand reg:
+                    DisposeRelatedPullReplication(reg.HubDefinitionName, reg.CertThumbprint);
+                    break;
+            }
+            return Task.CompletedTask;
+
+            void DisposeRelatedPullReplication(string hub, string certThumbprint)
+            {
+                if (hub == null)
+                    return;
+                
+                foreach (var  (_, repl) in _incoming)
+                {
+                    if (string.Equals(repl.PullReplicationName, hub, StringComparison.OrdinalIgnoreCase) == false) 
+                        continue;
+                    
+                    if(certThumbprint!= null && repl.CertificateThumbprint != certThumbprint)
+                        continue;
+                    
+                    try
+                    {
+                        if(_log.IsInfoEnabled)
+                            _log.Info("Resetting ");
```

Comment:
Can we have a bit more meaningful message here?

Reply:
fixed